### ISSUE TITLE
Array graphs outside graph-on-parent canvas when using [array set]

### ISSUE THREAD

**Issue by liliantdn** [pure-data/pure-data#1569]
_(I discovered this issue in pd-L2ork but after some testing it looks like it is also present in vanilla)_
When you update an array contained in a subpatch or abstraction that has graph-on-parent enabled using [array set], it will graph the array on the parent patch, even if the said array is outside of the graph-on-parent bounds. Here is a [video](https://streamable.com/psjkgk) showing this issue more clearly

**Comment by umlaeute** [comment]
could you provide a simple patch that shows the problem? this would be much better than a video...

**Comment by umlaeute** [comment]
also: how is `[array set]` involved?
does the problem appear when using other methods of setting an array (`[tabwrite]`, or just using `[; array1 0 0 5 8 9 10(`) ?

**Comment by liliantdn** [comment]
Yes sorry I should have provided a patch, here you go [array_graph-on-parent_issue.zip](https://github.com/pure-data/pure-data/files/8004054/array_graph-on-parent_issue.zip) . It seems like `[tabwrite]` and `[; array1 0 8 6 4 7...(` do not cause the same issue as `[array set]` (as you can see in the provided patch)
